### PR TITLE
fix: Return diagnostics on failed permissions in storage.Buckets

### PR DESCRIPTION
**Pull request by roneli**

### CODE REVIEW

File: client/error.go [hunk 1]
```diff
@@ -71,7 +71,13 @@ func ErrorClassifier(meta schema.ClientMeta, resourceName string, err error) dia
...
 			if v, ok := grpcCodeToDiag[grpcCode]; ok {
 				return diag.Diagnostics{
-					RedactError(client.projects, diag.NewBaseError(err, v.typ, diag.WithType(v.typ), diag.WithResourceName(resourceName), diag.WithSummary("%s", v.summary), diag.WithError(errors.New(gerr.Message)), diag.WithNoOverwrite(), diag.WithSeverity(v.severity))),
+					RedactError(client.projects,
+						diag.NewBaseError(err, v.typ,
+							diag.WithType(v.typ),
+							diag.WithResourceName(resourceName),
+							diag.WithSummary("%s", v.summary),
+							diag.WithError(errors.New(gerr.Message)),
+							diag.WithSeverity(v.severity))),
```
Comment (disq) [inline client/error.go:80]: removing `WithNoOverwrite` breaks things, now severity can't be overridden from IgnoreError.

Comment (roneli) [reply client/error.go:80]: But not removing this made the severity error, let me double check, as permission denied is always ignored error

File: resources/services/storage/storage_buckets.go [hunk 2]
```diff
@@ -578,13 +578,9 @@ func resolveBucketPolicy(ctx context.Context, meta schema.ClientMeta, resource *
 		return fmt.Errorf("expected *storage.Bucket but got %T", p)
 	}
 	cl := meta.(*client.Client)
-	call := cl.Services.Storage.Buckets.GetIamPolicy(p.Name)
+	call := cl.Services.Storage.Buckets.GetIamPolicy(p.Name).Fields("etag").OptionsRequestedPolicyVersion(3)
```
Comment (bbernays) [inline resources/services/storage/storage_buckets.go:581]: What about the field `bindings` ? Or how did you choose "etag"?

**Comment by disq** [comment]
Merge after tagging new SDK.

**Review by disq — APPROVED** [review]

**Review by bbernays — APPROVED** [review]
LGTM

**Pull request merged**